packages/ai-jsx/src/batteries/sidekick/platform/conversation.tsx

```diff
@@ -136,9 +161,33 @@ export function getNextConversationStep(
     case 'system':
     case 'user':
     case 'functionResponse': {
-      const generation = (
-        <ChatCompletion functionDefinitions={tools ? updatedTools : undefined}>{shrinkableConversation}</ChatCompletion>
+      let generation = (
+        <ChatCompletion functionDefinitions={hasTools ? updatedTools : undefined}>
+          {shrinkableConversation}
+        </ChatCompletion>
       );
+
+      const lastAssistantMessage = messages.findLast((m) => m.type === 'assistant');
+      if (
+        hasTools &&
```

Comment:
Nit: Why would we get a functionResponse if there are no tools?
And if we somehow get it, it's probably safe to interject still :?

Reply:
You're right, I don't think there's any need to make this conditional on tools.